Docs/ROADMAP.md

```diff
@@ -0,0 +1,328 @@
+- New: Go to new game menu
+- Settings: Go to settings menu
+- Credits: Go to credits menu
+- Quit: Exit the game
+
+#### Settings Menu
+
+This menu provides some settings
+
+- Master Volume: Volume for all sound 
+- Sound Effects Volume: Volume for sound effects
+- Music Volume: Volume for in-game music
+- Mouse sensitivity X: Sensitivity for turning left and right
+- Mouse Sensitivity y: Sensitivity for looking up and down
+- Name: Player's name
+
+There are also buttons on this menu:
+Main Menu: Return to main menu
+Revert: reload saved settings
+Save: Save current settings
+Controls: Navigate to controls menu
+
+#### Controls menu
+
+This menu provides various actions and the inputs they are mapped to.
+The user can click an action, then a new key to edit the mapping.
+
+- There are then some buttons on this menu:
+- Back: returns to options menu
+- Revert: load saved settings
+- Save: save current settings
+
+#### Credits menu
+
+A list of credits identifies all contributors and any special thanks.
+
+There is additionally a single button to return to the main menu.
+
+#### New game menu
+
+This menu shows the selection of the three starting characters. The player should be able to select each character to see information about them.
+
+There are buttons on this page:
+- Character selection buttons: When pressed highlight character and display information.
+- Main menu: Return to main menu
+- Start game: Appears when a character is selected. Creates a new save and navigates to the Career map.
+
+#### Career Menu
+
+This menu displays the player's progression through their career in the arena as an inverted tree of encounters.
+The tree's root node is at the top level of the menu, and its leaves are at the bottom of the menu.
+Each encounter is represented by a button that is disabled until active.
+To start with, only the leaf nodes of this tree on the bottom level are active. After playing through one encounter
+on a level, the parent nodes of that encounter become active. This is functionally identical to the map in Slay The Spire.
+The entire tree is randomly generated with different nodes upon the start of a new game.
+
+If the map has more levels than will fit on the screen, the player can scroll vertically by moving their mouse to the top or
+bottom center of the screen.
+
+There is also a button to return to the main menu.
+
+#### Rest Site Encounter Menu
+
+This menu initially displays two buttons
+- Rest button: Restores some health and completes encounter
+- Upgrade button: Display new set of buttons
+
+Once an upgrade is selected, the following buttons are displayed:
+- Buttons for up to three random upgrades
+- Confirmation button: Appears once an upgrade is selected. If clicked, upgrade is applied and encounter is completed
+
+#### Press Event Encounter Menu
+
+This menu displays a random prompt and up to four options for responses. 
+Some of these options have side effects(such as gaining money or upgrades or losing health)
```

Comment:
Do these side effects show up as like buff tiles or have some indicator in the game?

Reply:
That's still quite TBD. The current thought is to represent items, abilities, and upgrades as entities equivalent to the relics, potions, and cards found in Slay The Spire.